File: agent/bindings/csharp/ProtocolLocation.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolLocation
  {
    [JsonProperty(PropertyName = "uri")]
    public string Uri { get; set; }
    [JsonProperty(PropertyName = "range")]
    public Range Range { get; set; }
  }
}

File: vscode/src/jsonrpc/agent-protocol-factory/protocolLocation.ts
import * as vscode from 'vscode'
import type * as protocol from '../agent-protocol'
import { ProtocolRange, Uri } from './internal'

export namespace ProtocolLocation {
    export function from(location: vscode.Location): protocol.ProtocolLocation {
        return {
            uri: Uri.from(location.uri),
            range: ProtocolRange.from(location.range),
        }
    }

    export function vsc(location: protocol.ProtocolLocation): vscode.Location {
        return new vscode.Location(Uri.vsc(location.uri), ProtocolRange.vsc(location.range))
    }
}
